astropy/io/fits/src/compressionmodule.c

```diff
@@ -1063,6 +1063,24 @@ PyObject* compression_decompress_hdu(PyObject* self, PyObject* args)
     if (PyErr_Occurred()) {
         return NULL;
     }
+    // non-integer scaling or subtractive dithering means floating-point
+    if (floor(fileptr->Fptr->cn_bscale) != fileptr->Fptr->cn_bscale ||
+        floor(fileptr->Fptr->cn_bzero) != fileptr->Fptr->cn_bzero ||
+        fileptr->Fptr->
+#ifdef CFITSIO_SUPPORTS_SUBTRACTIVE_DITHER_2
+            quantize_method
```

Comment:
personall I would prefer to put the whole statement in a `ifdef`, just so it's more readable:

```
#ifdef CFITSIO_SUPPORTS_SUBTRACTIVE_DITHER_2
            fileptr->Fptr->quantize_method != NO_DITHER
#else
            fileptr->Fptr->quantize_dither != NO_DITHER
#endif
```